odl/operator/fn_ops.py

```diff
@@ -0,0 +1,406 @@
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+#
+#"""Default operators defined on any (reasonable) space."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+from odl.operator.operator import Operator
+from odl.space import fn
+from odl.set import LinearSpace, Field
+#from odl.util import writable_array
+
+import numpy as np
+
+__all__ = ('SamplingOperator', 'ZerofillingOperator', 
+           'VectoriseOperator', 'EmbeddingOperator')
+
+class SamplingOperator(Operator):
+
+    """Operator that samples coefficients.
+
+        ``SamplingOperator(f)(i) == cell_volume * f(i) \approx \int_cell f(x) dx``
+    """
+
+    def __init__(self, domain, indices):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        domain : `LinearSpace` or `Field`
+            Set of elements on which this operator acts.
+        indices : list
+            List of indices where to sample the function
+
+        Examples
+        --------
+        >>> X = odl.uniform_discr(min_pt=[0, 0], max_pt=[1, 1], shape=[2, 2])
+        >>> x = X.element(range(X.size))
+        >>> ind = list()
+        >>> ind.append(range(2))
+        >>> ind.append([1,1])
+        >>> A = odl.operator.SamplingOperator(X, ind)
+        >>> A(x)
+        rn(2).element([0.25, 0.75])
+        >>> A(x)  # Out-of-place
+        rn(2).element([0.25, 0.75])
+        """
+        if not isinstance(domain, (LinearSpace, Field)):
+            raise TypeError('`space` {!r} not a `LinearSpace` or `Field` '
+                            'instance'.format(domain))
+
+        self.__indices = np.asarray(indices, dtype=int)
+        # Converts a single sequence to an array of integers
+        # and a sequence of arrays to a vertically stacked array
+        if self.indices.ndim > 1:
+            # Strides = increment in linear indices per axis
+            strides = np.concatenate((np.cumprod(domain.shape[1:])[::-1], [1]))
+            self.__flat_indices = np.sum(self.indices * strides[:, None], axis=0)
+        
+        range = fn(self.flat_indices.size, dtype=domain.dtype)
+
+        super().__init__(domain, range, linear=True)
+
+    @property
+    def indices(self):
+        """indices where to sample the function."""
+        return self.__indices
+
+    @property
+    def flat_indices(self):
+        """indices where to sample the function."""
+        return self.__flat_indices
+
+    def _call(self, x, out=None):
+        """Collect indices weighted with the cell volume."""
+        c = getattr(self.domain, 'cell_volume', 1.0) # TODO: This only works for a subset of function spaces
+        if out is None:
+            out = x[self.flat_indices]*c
```

Comment:
This would fail [PEP8](https://www.python.org/dev/peps/pep-0008/) and should be written as

```
out = x[self.flat_indices] * c
```

To verify that your code follows pep8, you can run the whole test suite:

```
$ pytest --pep8
```